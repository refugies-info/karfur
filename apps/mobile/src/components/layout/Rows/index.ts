export { RowsSpacing, default } from "./Rows";
