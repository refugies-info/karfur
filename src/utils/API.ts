import axios from "react-native-axios";
import { getEnvironment } from "../libs/getEnvironment";

const dbURL = getEnvironment().dbUrl;
const siteSecret = getEnvironment().siteSecret;

const apiCaller = axios.create({
  baseURL: dbURL,
  headers: {
    "Content-Type": "application/json",
    "site-secret": siteSecret,
  },
});

export const getLanguages = () => apiCaller.get("/langues/getLanguages");

export const getCitiesFromGoogleAPI = (text: string) =>
  apiCaller.post(
    `https://maps.googleapis.com/maps/api/place/autocomplete/json?input=${text}&language=fr&types=(cities)&components=country:fr&key=${process.env.GOOGLE_API_KEY}`
  );

export const getCityDetailsFromGoogleAPI = (placeId: string) =>
  apiCaller.post(
    `https://maps.googleapis.com/maps/api/place/details/json?place_id=${placeId}&fields=address_component&key=${process.env.GOOGLE_API_KEY}`
  );

export const getPlaceIdFromLocationFromGoogleAPI = (
  longitude: number,
  latitude: number
) =>
  apiCaller.post(
    `https://maps.googleapis.com/maps/api/geocode/json?latlng=${latitude},${longitude}&key=${process.env.GOOGLE_API_KEY}`
  );

export const getContentsForApp = ({
  locale,
  age,
  department,
  frenchLevel,
}: {
  locale: string;
  age: string | null;
  department: string | null;
  frenchLevel: string | null;
}) => {
  const route =
    `/dispositifs/getContentsForApp?locale=${locale}` +
    (age ? `&age=${age}` : "") +
    (department ? `&department=${department}` : "") +
    (frenchLevel ? `&frenchLevel=${frenchLevel}` : "");

  return apiCaller.get(route);
};

export const get_dispositif = (params = {}) =>
  apiCaller.post("/dispositifs/get_dispositif", params);
