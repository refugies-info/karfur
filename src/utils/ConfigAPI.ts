import axios from "react-native-axios";
import Config from "../libs/getEnvironment";

const dbURL = Config.dbUrl;
const siteSecret = Config.siteSecret;
export const apiCaller = axios.create({
  baseURL: dbURL,
  headers: {
    "Content-Type": "application/json",
    "site-secret": siteSecret,
  },
});