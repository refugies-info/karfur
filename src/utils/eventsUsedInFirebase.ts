export const events = {
  REACHED_END_CONTENT: "REACHED_END_CONTENT",
  LANGUAGE_VALIDATE: "LANGUAGE_VALIDATE",
  LOCATION_VALIDATE: "LOCATION_VALIDATE",
};
