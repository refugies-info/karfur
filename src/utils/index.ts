export { default as withProps } from "./withProps";
