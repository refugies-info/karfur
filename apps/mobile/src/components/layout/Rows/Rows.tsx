/* eslint-disable react/require-default-props */ // props are optionnal
import isNull from "lodash/isNull";
import React, { ReactNode } from "react";
import { StyleProp, ViewStyle } from "react-native";
import styled from "styled-components/native";
import { isLastChild } from "../../utils";
import { FlexItem, getFlexValue } from "../common";
import Separator from "../Separator";

const RowsWrapper = styled.View<{
  horizontalAlign?: string;
  verticalAlign?: string;
}>`
  flex-grow: 1;
  flex-basis: auto;
  flex-direction: column;
  ${({ verticalAlign }) => verticalAlign && `justify-content: ${verticalAlign}`};
  ${({ horizontalAlign }) => horizontalAlign && `align-items: ${horizontalAlign}`};
`;

export enum RowsSpacing {
  Default = "default",
  NoSpace = "nospace",
  Text = "text",
}

export interface RowsProps {
  children: ReactNode;
  horizontalAlign?: string;
  verticalAlign?: string;
  layout?: string;
  spacing?: RowsSpacing;
  separator?: boolean;
  style?: StyleProp<ViewStyle>;
}

const Rows = ({
  children,
  horizontalAlign,
  verticalAlign,
  layout = "auto",
  spacing = RowsSpacing.Default,
  style,
  separator = false,
}: RowsProps) => {
  const _children = React.Children.toArray(children).filter((child: ReactNode) => !isNull(child));
  return (
    <RowsWrapper
      horizontalAlign={horizontalAlign}
      verticalAlign={verticalAlign}
      //@ts-ignore
      style={style}
    >
      {React.Children.map(
        _children,
        (child: ReactNode, index: number) =>
          child && (
            <FlexItem
              flex={getFlexValue(layout, index)}
              key={index}
              marginBottom={!isLastChild(_children, index) ? spacing : undefined}
            >
              {child}
              {separator && !isLastChild(_children, index) && <Separator fullWidth />}
            </FlexItem>
          ),
      )}
    </RowsWrapper>
  );
};

Rows.displayName = "Rows";

export default Rows;
