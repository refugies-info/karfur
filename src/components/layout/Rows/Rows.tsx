/* eslint-disable react/require-default-props */ // props are optionnal
import React, { ReactNode } from "react";
import { ViewStyle, StyleProp } from "react-native";
import styled from "styled-components/native";
import { FlexItem, getFlexValue } from "../common";
import { isLastChild } from "../../utils";
import { isNull } from "lodash";
import Separator from "../Separator";

const RowsWrapper = styled.View<{
  horizontalAlign?: string;
  verticalAlign?: string;
}>`
  flex-grow: 1;
  flex-basis: auto;
  flex-direction: column;
  ${({ verticalAlign }) =>
    verticalAlign && `justify-content: ${verticalAlign}`};
  ${({ horizontalAlign }) =>
    horizontalAlign && `align-items: ${horizontalAlign}`};
`;

export enum RowsSpacing {
  Default = "default",
  NoSpace = "nospace",
  Text = "text",
}

export interface RowsProps {
  children: ReactNode;
  horizontalAlign?: string;
  verticalAlign?: string;
  layout?: string;
  spacing?: RowsSpacing;
  separator?: boolean;
  style?: StyleProp<ViewStyle>;
}

const Rows = ({
  children,
  horizontalAlign,
  verticalAlign,
  layout = "auto",
  spacing = RowsSpacing.Default,
  style,
  separator = false,
}: RowsProps) => {
  const _children = React.Children.toArray(children).filter(
    (child: ReactNode) => !isNull(child)
  );
  return (
    <RowsWrapper
      horizontalAlign={horizontalAlign}
      verticalAlign={verticalAlign}
      //@ts-ignore
      style={style}
    >
      {React.Children.map(
        _children,
        (child: ReactNode, index: number) =>
          child && (
            <FlexItem
              flex={getFlexValue(layout, index)}
              // eslint-disable-next-line react/no-array-index-key
              key={index}
              marginBottom={!isLastChild(_children, index) && spacing}
            >
              {child}
            </FlexItem>
          )
      )}
    </RowsWrapper>
  );
};

Rows.displayName = "Rows";

export default Rows;
