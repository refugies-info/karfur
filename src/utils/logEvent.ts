/* eslint-disable no-console */
import * as Analytics from "expo-firebase-analytics";
import { getEnvironment } from "../libs/getEnvironment";
import { FirebaseEvent } from "./eventsUsedInFirebase";

export const logEventInFirebase = async (
  eventName: FirebaseEvent,
  data: Record<string, any>
) => {
  const { envName, debugModeFirebase } = getEnvironment();
  if (envName === "DEVELOPMENT") {
    if (debugModeFirebase === "activated") {
      Analytics.setDebugModeEnabled(true);
      console.log(
        "Environment is dev and debug mode is activated for firebase"
      );
    } else {
      console.log(
        "Environment is dev and debug mode is NOT activated for firebase"
      );
      return;
    }
  }

  if (envName && ["DEVELOPMENT", "STAGING", "PROD"].includes(envName)) {
    await Analytics.logEvent(eventName, { ...data, envName });
  }
};
