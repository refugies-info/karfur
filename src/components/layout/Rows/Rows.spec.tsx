import React from "react";
import { Text } from "react-native";
import Rows from "./Rows";
import { render } from "../../utils/tests";

describe("Rows snapshot test suite", () => {
  it("should render without bug", () => {
    const test = render(
      <Rows>
        <Text>Test</Text>
        <Text>Test</Text>
        <Text>Test</Text>
        <Text>Test</Text>
      </Rows>
    );
    expect(test).toMatchSnapshot();
  });

  it("should be able to control flex layout", () => {
    const test = render(
      <Rows layout="1 2" horizontalAlign="center" verticalAlign="space-between">
        <Text>Test</Text>
        <Text>Test</Text>
        <Text>Test</Text>
        <Text>Test</Text>
      </Rows>
    );
    expect(test).toMatchSnapshot();
  });
});
