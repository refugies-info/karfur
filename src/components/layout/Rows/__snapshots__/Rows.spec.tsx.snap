// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rows snapshot test suite should be able to control flex layout 1`] = `
<View
  horizontalAlign="center"
  style={
    [
      {
        "alignItems": "center",
        "flexBasis": "auto",
        "flexDirection": "column",
        "flexGrow": 1,
        "justifyContent": "space-between",
      },
    ]
  }
  verticalAlign="space-between"
>
  <View
    style={
      [
        {
          "flexBasis": 0,
          "flexGrow": 1,
          "marginBottom": 16,
        },
      ]
    }
  >
    <Text>
      Test
    </Text>
  </View>
  <View
    style={
      [
        {
          "flexBasis": 0,
          "flexGrow": 2,
          "marginBottom": 16,
        },
      ]
    }
  >
    <Text>
      Test
    </Text>
  </View>
  <View
    style={
      [
        {
          "flexBasis": 0,
          "flexGrow": 1,
          "marginBottom": 16,
        },
      ]
    }
  >
    <Text>
      Test
    </Text>
  </View>
  <View
    style={
      [
        {
          "flexBasis": 0,
          "flexGrow": 2,
        },
      ]
    }
  >
    <Text>
      Test
    </Text>
  </View>
</View>
`;

exports[`Rows snapshot test suite should render without bug 1`] = `
<View
  style={
    [
      {
        "flexBasis": "auto",
        "flexDirection": "column",
        "flexGrow": 1,
      },
    ]
  }
>
  <View
    style={
      [
        {
          "flexBasis": "auto",
          "flexGrow": 0,
          "marginBottom": 16,
        },
      ]
    }
  >
    <Text>
      Test
    </Text>
  </View>
  <View
    style={
      [
        {
          "flexBasis": "auto",
          "flexGrow": 0,
          "marginBottom": 16,
        },
      ]
    }
  >
    <Text>
      Test
    </Text>
  </View>
  <View
    style={
      [
        {
          "flexBasis": "auto",
          "flexGrow": 0,
          "marginBottom": 16,
        },
      ]
    }
  >
    <Text>
      Test
    </Text>
  </View>
  <View
    style={
      [
        {
          "flexBasis": "auto",
          "flexGrow": 0,
        },
      ]
    }
  >
    <Text>
      Test
    </Text>
  </View>
</View>
`;
